src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -343,39 +345,35 @@ await IsValidEndpoint(host).ConfigureAwait(false))
             _lastFallbackClientRefresh = DateTime.UtcNow;
         }
 
-        private async Task<bool> IsValidEndpoint(string hostName)
+        private string GetValidDomain(Uri endpoint)
         {
-            Debug.Assert(!string.IsNullOrEmpty(hostName));
+            Debug.Assert(endpoint != null);
+
+            string hostName = endpoint.Host;
 
-            if (_validDomains == null || !_validDomains.Any())
+            foreach (string domainPrefix in TrustedDomainPrefixes)

```

Comment:
The search needs to be from right to left.

Reply:
Seems to be fine since you look for segment prefixed with label separator.

Reply:
I'd still feel better if we went from right to left and found exact label match.

Reply:
Perhaps `.LastIndexOf($".{label}.")`

Reply:
Got you, I will update

Reply:
But we have a domain in such a pattern right? `azconfig-test.io`  Can we make exact label match?  Or we list all labels we have (or potentially have in the future) and traverse them?

Reply:
We'll have to think about azconfig-test.io. Most likely, we'll have to phase it out and adopt a proper .azconfig.xyz domain in ppe.

Reply:
Updated